### ISSUE TITLE
Keep getting `SSLError: SSL_connect ... wrong version number`

### ISSUE THREAD

**Issue by niekvb** [mastodon/mastodon#22235]
### Steps to reproduce the problem

1. Install Mastodon using Helm chart in Kubernetes.
2. Modify environment variables and restart pods.
3. Find failed email jobs in Sidekiq.
4. Try to restart jobs.
5. Modify env vars again and repeat at 3.

### Expected behaviour

Email jobs to succeed

### Actual behaviour

Sending email fails with errors below

### Detailed description

Following exception in Sidekiq pod:
```plaintext
2022-12-11T07:18:49.249Z pid=1 tid=2f7h class=ActionMailer::MailDeliveryJob jid=619fd781c4ebda9741bde56e elapsed=0.296 INFO: fail
2022-12-11T07:18:49.249Z pid=1 tid=2f7h WARN: {"context":"Job raised exception","job":{"retry":true,"queue":"mailers","class":"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper","wrapped":"ActionMailer::MailDeliveryJob","args":[{"job_class":"ActionMailer::MailDeliveryJob","job_id":"d928b376-b000-4339-ad04-e3696a2b62e4","provider_job_id":null,"queue_name":"mailers","priority":null,"arguments":["UserMailer","two_factor_enabled","deliver_now!",{"args":[{"_aj_globalid":"gid://mastodon/User/1"}],"_aj_ruby2_keywords":["args"]}],"executions":0,"exception_executions":{},"locale":"en","timezone":"UTC","enqueued_at":"2022-12-11T01:26:09Z"}],"jid":"619fd781c4ebda9741bde56e","created_at":1670721969.8487694,"enqueued_at":1670743128.9527729,"error_message":"SSL_connect returned=1 errno=0 peeraddr=xxx.xxx.xxx.xxx:587 state=error: wrong version number","error_class":"OpenSSL::SSL::SSLError","failed_at":1670721970.1694205,"retry_count":8,"retried_at":1670742780.7399924}}
2022-12-11T07:18:49.249Z pid=1 tid=2f7h WARN: OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 peeraddr=xxx.xxx.xxx.xxx:587 state=error: wrong version number
2022-12-11T07:18:49.250Z pid=1 tid=2f7h WARN: /opt/ruby/lib/ruby/3.0.0/net/protocol.rb:46:in `connect_nonblock'
/opt/ruby/lib/ruby/3.0.0/net/protocol.rb:46:in `ssl_socket_connect'
/opt/ruby/lib/ruby/3.0.0/net/smtp.rb:624:in `tlsconnect'
```

The following environment variables are set:
```plaintext
SMTP_SERVER=mail.domain.example
SMTP_ENABLE_STARTTLS=auto
SMTP_TLS=true
SMTP_PORT=587
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_LOGIN=mastodon@domain.example
SMTP_AUTH_METHOD=plain
SMTP_DELIVERY_METHOD=smtp
SMTP_DOMAIN=domain.example
SMTP_PASSWORD=someverylongpasswordhasbeensetheresomewhereinsteadofthissentence
SMTP_FROM_ADDRESS=mastodon@domain.example
SMTP_REPLY_TO=admin@domain.example
```

The following is logged on the mailserver:
```plaintext
postfix/submission/smtpd[17795]: connect from unknown[10.1.39.66]
postfix/submission/smtpd[17795]: lost connection after UNKNOWN from unknown[10.1.39.66]
postfix/submission/smtpd[17795]: disconnect from unknown[10.1.39.66] unknown=0/3 commands=0/3
```

I scoured the internet and issues listed here but didn't find a working solution... I also tried all possible combinations of env vars for setting up email securely but without any luck.

I know the config above (in env vars) works as it is used by several other applications.

### Specifications

Mastodon: v4.0.2
Sidekiq 6.5.7
Rails 6.1.7
Ruby 3.0.4p208

**Comment by ineffyble** [comment]
Try setting `SMTP_TLS=false`.

Port 587 usually means "use STARTTLS" instead of "use TLS".

**Comment by niekvb** [comment]
@ineffyble I tried that as well, but without luck. Thanks for the tip though.

Inspection of the mailserver logs indicates that Sidekiq communicates to the mailserver container directly (due to the IP listed in the 10.1.x.x range) instead of other services on my cluster do as they log the node's external IP. So it resolves `mail.domain.example` to an internal IP.

In this case the issue is because of this issue I posted here: https://github.com/docker-mailserver/docker-mailserver/issues/2929

I'll close this for now as it probably has to do with the issue mentioned above.
